File: server/Models/GreenhouseHarvestItem.cs
namespace Server.Models;

/// <summary>
/// How much of a <see cref="GreenhouseStock"/> good is planned for a <see cref="GreenhouseHarvest"/>.
/// The greenhouse counterpart to <see cref="HarvestItem"/> — always points at greenhouse stock,
/// since a greenhouse has no orchard equivalent to pick from.
/// </summary>
public class GreenhouseHarvestItem
{
    public int Id { get; set; }
    public int GreenhouseHarvestId { get; set; }
    public int GreenhouseStockId { get; set; }
    public decimal Amount { get; set; }

    /// <summary>
    /// The unit <see cref="Amount"/> is planned in. A plan may use a different unit than the good
    /// is stocked in, so it is stored per row rather than read off the target. Left blank by a
    /// client, it falls back to the target's own unit — see GreenhouseHarvestItemRepository.
    /// </summary>
    public string Unit { get; set; } = string.Empty;
}

File: web/src/types/greenhouse-harvest-item.ts
/** How much of a GreenhouseStock good is planned for a greenhouse harvest — always points at
 * greenhouse stock, since a greenhouse has no orchard equivalent to pick from. */
export type GreenhouseHarvestItem = {
  id: number;
  greenhouseHarvestId: number;
  greenhouseStockId: number;
  amount: number;
  /** The unit the plan is expressed in. May differ from the unit the target good is stocked in. */
  unit: string;
};

export type GreenhouseHarvestItemInput = Omit<GreenhouseHarvestItem, 'id'>;
